[ews-build.webkit.org] Convert LeaveComment to Twisted asynchronous request

JonWBedard opened the pull request:
#### 26375ba46ade830e95152f2b8a75d094ccb66721
<pre>
[ews-build.webkit.org] Convert LeaveComment to Twisted asynchronous request
<a href="https://bugs.webkit.org/show_bug.cgi?id=251589">https://bugs.webkit.org/show_bug.cgi?id=251589</a>
rdar://104961332

Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/steps.py:
(GitHubMixin.comment_on_pr): Convert to Twisted asynchronous requests.
(BugzillaMixin.comment_on_bug): Ditto.
(LeaveComment.run): Handle defered results from comment_on_pr and comment_on_bug.

Canonical link: <a href="https://commits.webkit.org/259770@main">https://commits.webkit.org/259770@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bfe4d4bbb1e240696c512b0b7035a6b9e0fc7c60

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [  ~~🧪 style~~](https://ews-build.webkit.org/#/builders/6/builds/105890 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/77/builds/14955 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/38738 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/8/builds/115086 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/175207 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [  ~~🧪 bindings~~](https://ews-build.webkit.org/#/builders/11/builds/109789 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/76/builds/16377 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/85/builds/6155 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/98120 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/114845 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/19/builds/111639 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/76/builds/16377 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/38738 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/98120 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/76/builds/16377 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/38738 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/98120 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/81/builds/8225 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/38738 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/82/builds/8706 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/85/builds/6155 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [  ~~🧪 services~~](https://ews-build.webkit.org/#/builders/20/builds/104689 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/14336 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/38738 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/79/builds/10261 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [❌ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3623 "Failed to push commit to Webkit repository") | | | | 
<!--EWS-Status-Bubble-End-->

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -389,27 +389,34 @@ def remove_labels(self, pr_number, labels=None, repository_url=None):
-                json=dict(body=content),
+            if username and access_token:
+                auth_header = b64encode('{}:{}'.format(username, access_token).encode('utf-8')).decode('utf-8')
+                headers['Authorization'] = ['Basic {}'.format(auth_header)]
```
can use f-string here

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -389,27 +389,34 @@ def remove_labels(self, pr_number, labels=None, repository_url=None):
-                headers=dict(Accept='application/vnd.github.v3+json'),
-                json=dict(body=content),
+            if username and access_token:
+                auth_header = b64encode('{}:{}'.format(username, access_token).encode('utf-8')).decode('utf-8')
```
can use f-string here.

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -389,27 +389,34 @@ def remove_labels(self, pr_number, labels=None, repository_url=None):
-            self._addToLog('stdio', f"Error in posting comment to PR {pr_number}\n")
-            return False
-        return True
+            yield self._addToLog('stdio', f"Error in posting comment to PR {pr_number}\n")
```
might be good idea to print `e` as well

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -1561,19 +1568,26 @@ def close_bug(self, bug_id):
             if response.status_code not in [200, 201]:
-                self._addToLog('stdio', 'Unable to comment on bug {}. Unexpected response code from bugzilla: {}'.format(bug_id, response.status_code))
-                return FAILURE
+                yield self._addToLog('stdio', 'Unable to comment on bug {}. Unexpected response code from bugzilla: {}'.format(bug_id, response.status_code))
```
can use f-string here.

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -1561,19 +1568,26 @@ def close_bug(self, bug_id):
-            self._addToLog('stdio', 'Error in commenting on bug {}'.format(bug_id))
-            return FAILURE
-        return SUCCESS
+            yield self._addToLog('stdio', 'Error in commenting on bug {}'.format(bug_id))
```
can use f-string here. Also can print `e` as well.

aj062 reviewed (approved):
r+ assuming this is tested

JonWBedard committed:
[ews-build.webkit.org] Convert LeaveComment to Twisted asynchronous request
https://bugs.webkit.org/show_bug.cgi?id=251589
rdar://104961332

Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/steps.py:
(GitHubMixin.comment_on_pr): Convert to Twisted asynchronous requests.
(BugzillaMixin.comment_on_bug): Ditto.
(LeaveComment.run): Handle defered results from comment_on_pr and comment_on_bug.

Canonical link: https://commits.webkit.org/259770@main
